ships/0023-surface-results-in-buildruns.md

```diff
@@ -0,0 +1,304 @@
+
+#### Current State of Build Results for Sources
+
+Assuming the following Build definition:
+
+```yaml
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: a-build
+spec:
+  source:
+    url: https://github.com/shipwright-io/sample-go
+
+```
+
+We will have the following:
+
+| Result Name                       | Naming Convention                      | Emitted Today | Definition  | Notes                                |
+| --------------------------------: | ------------------------------------------: | -------: | ----------: | -----------------------------------: |
+| shp-source-default-commit-sha     |  `shp-source-${sourceName}-${resultName}`   | yes     |  at runtime | `${sourceName}` is `default` as `spec.source` has no name. |
+
+#### Future State of Build Results for Sources
+
+Assuming the following Build definition:
+
+```yaml
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: a-build
+spec:
+  sources:
+  - name: a-git-source
+    type: git
+    url: https://github.com/shipwright-io/sample-go
+  - name: an-http-source
+    type: http
+    url: https://licenses.company.com/customer-id/license.tar
+```
+
+We will have the following:
+
+| Result Name                       | Naming Convention                           | Emitted Today | Definition  |
+| --------------------------------: | ------------------------------------------: |-------: | ----------: |
+| shp-source-a-git-source-commit-sha     |  `shp-source-${spec.sources[0].name}-${resultName}` | no     |  at runtime |
+| shp-source-a-git-source-commit-author     |  `shp-source-${spec.sources[0].name}-${resultName}` | no     |  at runtime |
+| shp-source-an-http-source-size     |  `shp-source-${spec.sources[1].name}-${resultName}` | no     |  at runtime |
+
+#### Build Results Sources API Proposal
+
+Because of the future adoption of sources, the proposed API for `Build Results` should consider the following:
+
+- The _.status_ subresource should provide a relationship between a source and the results emitted in it´s step definition.
+- The _.status_ subresource should be able to surface results emitted from different sources.
+
+Therefore the **proposal** is to introduce `_.status.sources_` subresource which holds a list of `SourcesResult`:
+
+```go
+type BuildRunStatus struct { // This struct already exists on the BuildRun API, we just add a new field
+  Sources []SourceResult `json:"sources"`
+}
+
+type SourceResult struct {
+  Name   string             `json:"name"`
+  Git    *GitSourceResult   `json:"git,omitempty"`
+  Http   *HttpSourceResult  `json:"http,omitempty"` // A new field can be always introduced to support new types, e.g. bundle
+}
+
+type GitSourceResult struct{
+  CommitSha     string `json:"commit-sha,omitempty"`
+  CommitAuthor  string `json:"commit-author,omitempty"`
```

Comment:
```suggestion
  CommitAuthor  string `json:"commit-author,omitempty"` // Not existing today, but outlining how it would look with multiple results
```